aten/src/ATen/native/cuda/ForeachTensorAddList.cu

```diff
@@ -0,0 +1,205 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/cuda/ForeachUtils.cuh>
+#include <ATen/native/cuda/MultiTensorApply.cuh>
+
+namespace at { namespace native {
+
+namespace {
+
+template<typename x_t, typename y_t>
+struct AddListFunctor_
```

Comment:
I think it's worth splitting these Functors with a slim interface into another file to separate the wrapping code from the implementation (which might change in the future). This seems a bit too specific to multi_tensor_apply.